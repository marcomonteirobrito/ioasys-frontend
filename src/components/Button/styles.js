import styled from 'styled-components';

import Colors from '../../utils/Colors';

export const ButtonContainer = styled.button`
  width: 85px;
  height: 36px;
  margin: 12px;
  background: ${Colors.white};
  color: ${Colors.buttonLabelColor};
  border-radius: 44px;
  border: none;

  :hover {
    opacity: 0.9;
  }

  :disabled {
    cursor: default;
    color: #c7c7cc;
  }
`;
