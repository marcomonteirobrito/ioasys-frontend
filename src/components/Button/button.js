import React from 'react';
import PropTypes from 'prop-types';

import { ButtonContainer } from './styles';

/**
 * @param {string} className - ClassName do elemento
 * @param {function} onClick - Função a ser chamada no evento de click
 * @param {boolean} disabled - Indica se o botão deve ficar desabilitado
 * @param {boolean} loading - Indica se o botão deve apresentar loading
 * @param {string} children - Conteúdo dentro do botão
 */

const Button = ({ className, onClick, disabled, children }) => (
  <ButtonContainer
    className={className}
    type="button"
    disabled={disabled}
    onClick={onClick}
  >
    {children}
  </ButtonContainer>
);
Button.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  children: PropTypes.string,
};

Button.defaultProps = {
  className: 'button-container',
  disabled: false,
  children: 'Entrar',
};

export default Button;
