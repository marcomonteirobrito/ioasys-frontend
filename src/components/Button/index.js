import React from 'react';
import PropTypes from 'prop-types';

import Loading from '../Loading';
import { ButtonContainer } from './styles';

/**
 * @param {string} className - ClassName do elemento
 * @param {function} onClick - Função a ser chamada no evento de click
 * @param {boolean} disabled - Indica se o botão deve ficar desabilitado
 * @param {boolean} loading - Indica se o botão deve apresentar loading
 * @param {string} children - Conteúdo dentro do botão
 */

const Button = ({ className, onClick, disabled, loading, children }) =>
  loading ? (
    <Loading strokeWidth="5" />
  ) : (
    <ButtonContainer
      className={className}
      type="button"
      disabled={disabled}
      onClick={onClick}
    >
      {children}
    </ButtonContainer>
  );

Button.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  loading: PropTypes.bool,
  children: PropTypes.string,
};

Button.defaultProps = {
  className: 'button-container',
  disabled: false,
  children: 'Entrar',
  loading: false,
};

export default Button;
