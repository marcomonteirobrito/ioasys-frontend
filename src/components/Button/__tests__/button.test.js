/* eslint-disable no-undef */
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

import Button from '../button';

const renderButton = () => {
  const onButtonClick = jest.fn();
  return render(<Button onClick={onButtonClick}>Entrar</Button>);
};

describe('Button', () => {
  it('Deve renderizar o componente corretamente', () => {
    renderButton();

    expect(document.body).toMatchSnapshot();
  });

  it('Deve incluir children', () => {
    renderButton();

    const children = screen.queryByText('Entrar');

    expect(children).toBeDefined();
  });

  it('Deve chamar onClick ao clicar no botÃ£o', () => {
    const onButtonClick = jest.fn();
    render(<Button onClick={onButtonClick}>Entrar</Button>);

    const children = screen.queryByText('Entrar');
    fireEvent.click(children);

    expect(onButtonClick).toHaveBeenCalledTimes(1);
  });
});
