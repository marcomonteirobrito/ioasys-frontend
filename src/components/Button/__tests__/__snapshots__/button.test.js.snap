// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Deve renderizar o componente corretamente 1`] = `
<body>
  <div>
    <button
      class="sc-bdfBwQ ljibip button-container"
      type="button"
    >
      Entrar
    </button>
  </div>
</body>
`;
